pkg/virtctl/imageupload/imageupload_test.go

```diff
@@ -412,6 +412,46 @@ var _ = Describe("ImageUpload", func() {
 			validateBlockDataVolume()
 		})
 
+		It("Create a VolumeMode=Block PVC by specifying volume-mode", func() {
+			testInit(http.StatusOK)
+			cmd := tests.NewRepeatableVirtctlCommand(commandName, "dv", targetName, "--size", pvcSize,
```

Comment:
This is a slow e2e test. We have too many of these already. I'm looking for a simpler *unit* test for the new `normalizeVolumeMode` function. Something like

```
Describe("normalizeVolumeMode", func() {
  It("considers Filesystem as default", func() {
     volumeMode, err = normalizeVolumeMode("", "")
     Expect(err).To(BeNil()))
     Expect(volumeMode).To(Equal(string(v1.PersistentVolumeFilesystem)))
  }
  It("considers --block-volume=true as volumeMode=Block", func() {
     volumeMode, err = normalizeVolumeMode("", "true")
     Expect(err).To(BeNil()))
     Expect(volumeMode).To(Equal(string(v1.PersistentVolumeBlock)))
  }
})
```

notice that in my opinion, `blockVolume` should be a string that `normalizeVolumeMode` converts into `bool` and then to `v1.PersistentVolumeMode`.

Reply:
I think the convention is that we don't write tests for unexported function. 

>notice that in my opinion, blockVolume should be a string that normalizeVolumeMode converts into bool and then to v1.PersistentVolumeMode.

I don't understand why it's needed, it looks not necessary to me. 
Plus, the current code looks good without converting `blockVolume ` to string.

Reply:
> I think the convention is that we don't write tests for unexported function.

Testing a private function has its downsides, but forcing you to have an entire k8s system just to test your code is even worse. IMHO we should cover much more of our code with unit tests.

> 
> > notice that in my opinion, blockVolume should be a string that normalizeVolumeMode converts into bool and then to v1.PersistentVolumeMode.
> 
> I don't understand why it's needed, it looks not necessary to me.
> Plus, the current code looks good without converting `blockVolume ` to string.

The current code has a fuzzy semantics - it prefers the old-style flag to the new-style one if both show up, while it should be an error. I don't really like it, and prefer to improve it.